rust/hyperlane-base/src/settings/loader/arguments.rs

```diff
@@ -77,12 +86,15 @@ impl Source for CommandLineArguments {
                 continue;
             }
 
-            let mut key = key.to_lowercase();
-
-            // If separator is given replace with `.`
-            if !separator.is_empty() && separator != "." {
-                key = key.replace(separator, ".");
-            }
+            let key = if let Some(case) = self.casing {
+                // if case is given, replace case of each key component
+                key.split(separator).map(|s| s.to_case(case)).join(".")
```

Comment:
Seems like there's two things going on at the same time here:
- split tokens are being converted to `case`
- `separator` occurences are replaced with `"."`. This part is also handled by the `else if` branch

This would be easier to understand to me:
```rust
            // have `key` declared as mutable 
            if let Some(case) = self.casing {
                // if case is given, replace case of each key component
                key = key.split(separator).map(|s| s.to_case(case)).join(separator)
            }
            // If separator is given replace with `.`
            if !separator.is_empty() && separator != "." {
                key = key.replace(separator, ".")
            }
```

Reply:
I considered that, but it is better to avoid allocating the extra string so I will instead add some docs to clarify what is happening.